### PR TITLE
Add J# support

### PR DESCRIPTION
**Pull request by KrzysztofSzewczyk**
## Description

This PR is adding J# language support as stated [there](https://github.com/github/linguist/issues/4124). 

## Checklist:
- [X] **I am adding a new language.**
  - [X] The extension of the new language is used in hundreds of repositories on GitHub.com.
    (described more in linked issue)
  - [X] I have included a syntax highlighting grammar: Not necessary, using Java one.
  - [X] I have updated the heuristics to distinguish my language from others using the same extension: Other languages with .jsl extension are not warranted support by Linguist, therefore I don't mind them.

## My TODO list:
 - [X] Resolve color conflicts
 - [X] Move samples to valid directory

Also, I ran `update-ids` on languages.yml, though I don't think this is expected result.

A few repositories:
 * https://github.com/matt77hias/jsl-smallpt
 * https://github.com/ankurdave/htw-3d
 * https://github.com/hikerlive/code4me/
 * https://github.com/duchien85/cellengine/blob/4bafd127f579d3cdc1f652500111ad8b88588b8e/gameedit/CellMusicEdit/AppGameBMS/Program.jsl
 * Many, many, many more: https://github.com/search?l=&q=extension%3Ajsl+Program&type=Code
 * Even more: https://github.com/search?l=&q=extension%3Ajsl+class&type=Code

### CODE REVIEW

**Comment by KrzysztofSzewczyk** [comment]
Done, added five samples and removed crap update-ids made.

**Comment by KrzysztofSzewczyk** [comment]
@lildude @pchaigno Are there any other changes I need to make?

**Comment by KrzysztofSzewczyk** [comment]
@lildude @pchaigno Are there any other changes I need to make?

**Comment by lildude** [comment]
> Are there any other changes I need to make?

Yup. You need to get the tests passing. You can ignore the license related failures as these are unrelated, but the other failures are definitely related.

**Comment by KrzysztofSzewczyk** [comment]
@lildude check CI build log, everything is fine as I can see.
I didn't change anything other, I swear.

**Comment by KrzysztofSzewczyk** [comment]
@lildude @pchaigno everything except licenses is done

**Comment by KrzysztofSzewczyk** [comment]
@pchaigno Done

File: test/test_grammars.rb [hunk 1]
```diff
@@ -11,29 +11,27 @@ class TestGrammars < Minitest::Test
   ].freeze
 
   HASH_WHITELIST = [
-    "2edac46b0a63309c96442d2826321a442217472f", # Agda.tmbundle
-    "4da01d631a29c76456fd0bd16749c71e8d5f6dbf", # ant.tmbundle
-    "79e72fd673dcebadd8fbace8d43db3da96d2c09f", # bro-sublime
+    "4b4c1b4aff11b99e53091f95b24093655c0d9ed2", # Agda.tmbundle
+    "a92c0e4e644206078a158d663b9f033d0d0aefbf", # ant.tmbundle
+    "98eac91395c97ae090113e1b4527cae3228d563a", # bro-sublime
```
Comment (pchaigno) [inline test/test_grammars.rb]: These changes don't look related to your pull request. I guess you messed something up while updating your branch. Please remember to check what you are comtting before committing.

Comment (KrzysztofSzewczyk) [reply test/test_grammars.rb]: You told me to update my branch. @pchaigno

Comment (pchaigno) [reply test/test_grammars.rb]: Yes. It looks like it wasn't properly updated though, or changes to `test/test_grammars.rb` wouldn't appear. How did you update it?

Comment (KrzysztofSzewczyk) [reply test/test_grammars.rb]: @pchaigno by hand

Comment (pchaigno) [reply test/test_grammars.rb]: I mean what commands?

Comment (pchaigno) [reply test/test_grammars.rb]: Oh, I see. That's not how you update a branch :-)
Please see [this SO answer](https://stackoverflow.com/a/3877000/6884590) for two methods to update a branch (merge or rebase). I'd recommend you revert the changes you made and apply one of these methods.

Comment (KrzysztofSzewczyk) [reply test/test_grammars.rb]: @pchaigno I see, that repo was a little bit messy.
I've cloned fresh one and made only mine changes.

**Comment by KrzysztofSzewczyk** [comment]
#4257 was made.

**Comment by pchaigno** [comment]
You know you can rewrite an existing pull request by force pushing to the branch (`git push -f origin yourbranch`)?
